### DISCUSSION TITLE
Module -M meshing failed

### DISCUSSION THREAD

**Discussion by majidkavousi** [neperfepx/neper#450] [Q&A] [answered]
Hi,
I am getting some strange random errors while meshing some simple tessellations using module -M. 

For instance, I have generated the following mesh:
$ neper -T -n 20 -morpho gg -o sim2
![sim2](https://user-images.githubusercontent.com/99205979/164514520-6d2f0f4e-6d93-45c2-a21d-8f63f109d72a.png)

When I want to mesh it, I get an error as **Meshing of face 2 failed
Aborted (core dumped).** 
![image](https://user-images.githubusercontent.com/99205979/164516021-1358fbdb-333a-4eb0-a9c3-0e1020f34b1a.png)

While the specified face number varies every time I try meshing again. 

I tried to mesh with different element lengths (-rcl or -cl), again I get the same error with some random face number.
For the 2D case, I managed to mesh some simple tessellations successfully as below:
![test_mesh](https://user-images.githubusercontent.com/99205979/164515240-bd2f9b81-f11f-4f31-a6c3-917cd5847efd.png)
But, even in this case, if I run the meshing code again, I would get the same error on some faces. 
I think this issue is not there for quad or hex element types and these type of elements works fine presumably. 

Does anyone have an idea about this error?

**Comment by rquey** [comment, marked as answer]
Please do not copy paste text as images. Just copy paste as text.

I think that Gmsh is missing or not installed properly. The reason why you get a different face number each time is multithreading: several faces are meshed in parallel, and the first one to fail is reported. It can change from one run to the next.

**Comment by schakrabortygithub** [comment]
It seems you are using Windows Subsystem Linux.
On the introduction page of the documentation, it is mentioned that "Neper also runs on Microsoft Windows using the Windows subsystem for Linux (WSL), the Ubuntu app or similar solutions, although we do not test nor support this usage and **multithreading must be turned off** (which slows down execution significantly)."
You can try using a single thread. Use 'export OMP_NUM_THREADS=1' before running the meshing module.


![NEPER_multithreading](https://user-images.githubusercontent.com/35871741/174259035-f4aa380a-2904-46fc-8785-406f745355ce.JPG)
